src/lib.rs

```
@@ -152,7 +152,13 @@ pub unsafe fn load_from_path<P>(path: P) -> Result<Wintun, Error>
 where
     P: AsRef<::std::ffi::OsStr>,
 {
-    unsafe { Ok(Arc::new(wintun_raw::wintun::new(path)?)) }
+    let dll_path = path.as_ref().to_string_lossy().to_string();
```

Why change this? From what I see `to_string_lossy` will cause the dll to fail to load if the path is not utf8. Also since this is a library function I prefer to return the error and let the user handle logging